ablySpec/RealtimeClientChannel.swift

```diff
@@ -607,6 +607,70 @@ class RealtimeClientChannel: QuickSpec {
                     expect(Test.counter).toEventually(equal(2), timeout: testTimeout)
                 }
 
+                // RTL7d
+                pending("should deliver the message even if there is an error while decoding") {
+
+                    for cryptoTest in [CryptoTest.aes128, CryptoTest.aes256] {
+                        it("using \(cryptoTest) ") {
+                            let options = AblyTests.commonAppSetup()
+                            options.autoConnect = false
+                            let client = ARTRealtime(options: options)
+                            client.setTransportClass(TestProxyTransport.self)
+                            client.connect()
+                            defer { client.close() }
+
+                            let (keyData, ivData, messages) = AblyTests.loadCryptoTestData(cryptoTest)
+                            let testMessage = messages[0]
+
+                            let cipherParams = ARTCrypto.defaultParamsWithKey(keyData, iv: ivData)
+                            let channelOptions = ARTChannelOptions(encrypted: cipherParams)
+                            let channel = client.channels.get("test", options: channelOptions)
+
+                            let transport = client.transport as! TestProxyTransport
+
+                            transport.beforeProcessingSentMessage = { protocolMessage in
+                                if protocolMessage.action == .Message {
+                                    expect(protocolMessage.messages![0].data as? String).to(equal(testMessage.encrypted.data))
+                                    expect(protocolMessage.messages![0].encoding).to(equal(testMessage.encrypted.encoding))
+                                }
+                            }
+
+                            transport.beforeProcessingReceivedMessage = { protocolMessage in
+                                if protocolMessage.action == .Message {
+                                    expect(protocolMessage.messages![0].data as? String).to(equal(testMessage.encrypted.data))
+                                    expect(protocolMessage.messages![0].encoding).to(equal(testMessage.encrypted.encoding))
+                                    // Force an error decoding a message
+                                    protocolMessage.messages![0].encoding = "bad_encoding_type"
+                                }
+                            }
+
+                            waitUntil(timeout: testTimeout) { done in
+                                let logTime = NSDate()
+
+                                channel.on(.Failed) { errorInfo in
```

Comment:
I realise we are not emitting events here, but the channel should not become failed, so this test is wrong. We don't expect his behaviour.